### DISCUSSION TITLE
nio/BIONIOAIO

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#193] [Announcements]
# nio/BIONIOAIO

BIO采用阻塞式 I/O 模型，线程在执行 I/O 操作时被阻塞，无法处理其他任务，适用于连接数较少且稳定的场景。NIO使用非阻塞 I/O 模型，线程在等待 I/O 时可执行其他任务，通过 Selector 监控多个 Channel 上的事件，提高性能和可伸缩性，适用于高并发场景。AIO采用异步 I/O 模型，线程发起 I/O 请求后立即返回，当 I/O 操作完成时通过回调函数通知线程，进一步提高了并发处理能力，适用于高吞吐量场景。

https://tobebetterjavaer.com/nio/BIONIOAIO.html

**Comment by lige0** [comment]
Java NIO，全程 Non-Block IO ，是 Java SE 1.4 版以后，针对网络传输效能优化的新功能。是一种非阻塞同步的通信模式。
二哥，这里这个“全程”错了

**Comment by light-dot** [comment]
AIO中的readFile()方法中，为啥要用Thread.sleep(1000)？

**Comment by Aurorxa** [comment]
如果不从 Linux 系统编程以及操作系统那边讲，其实也只是在堆积代码。
